src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/config/InterceptorConfig.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.companieshouse.limitedpartnershipsapi.config;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.NonNull;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import uk.gov.companieshouse.limitedpartnershipsapi.interceptor.LoggingInterceptor;
+
+
+@Configuration
+public class InterceptorConfig implements WebMvcConfigurer {
+
+    private final LoggingInterceptor loggingInterceptor;
+
+    public InterceptorConfig(LoggingInterceptor loggingInterceptor) {
+        this.loggingInterceptor = loggingInterceptor;
+    }
+
+    /**
+     * Set up the interceptors to run against endpoints when the endpoints are called
+     * Interceptors are executed in the order they are added to the registry
+     * @param registry The spring interceptor registry
+     */
+    @Override
+    public void addInterceptors(@NonNull InterceptorRegistry registry) {
+        addLoggingInterceptor(registry);
+    }
+
+    /**
+     * Interceptor that logs all calls to endpoints
+     * @param registry The spring interceptor registry
+     */
+    private void addLoggingInterceptor(InterceptorRegistry registry) {
+        registry.addInterceptor(loggingInterceptor);
```

Comment:
Could just move this line to 26 and not have the private method (with Javadoc). Save on LOC... Or do you think it makes code more readable?

Reply:
i can move it